app/models/user.rb

```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  validates :email, presence: true, uniqueness: true
```

Comment:
Валидацию на формат email'а можно перенести на след строку у твоей первой валидации имейла, тем самым не будет лишней строчки
```suggestion
  validates :email, presence: true, uniqueness: true,
    format: { with: /\A[^@\s]+@[^@\s]+\z/, message: "Invalid email format" }
```
Твой вариант тоже верный, но предложенный вариант предпочтительнее
[ИМХО] 😄